File: 08FlowControl/match/guard.rs
fn main() {
    let pair = (3, -1);//匹配元组的情况
    // 试一试 ^ 将不同的值赋给 `pair`

    println!("Tell me about {:?}", pair);
    match pair { //元组，最多12项，取部分或者全部元素，然后根据类型条件来匹配。
        (x, y) if x == y => println!("These are twins"),
        // ^ `if` 条件部分是一个卫语句
        (x, y) if x + y == 0 => println!("Antimatter, kaboom!"),
        (x, _) if x % 2 == 1 => println!("The first one is odd"),
        _ => println!("No correlation..."),
    }
}

File: 08FlowControl/match/guard.cs
using System;
public class Program
{
    //guard,就是match 然后用if的方式。
    public static void Main(string[] args)
    {
        Console.WriteLine("This is C# Language");

        var pair = (3, -1);
        Console.WriteLine("Tell me about {0}", pair);

        var message = pair switch{
            (var x,var y) when x == y =>"These are twins",
            (var x,var y) when x + y == 0 =>"Antimatter, kaboom!",
            (var x, _) when x % 2 == 1=>"The first one is odd",
            _=>"No correlation...",
        };
        Console.WriteLine( message );
    }
}
/*Output
Tell me about (3, -1)
The first one is odd
*/